src/work-allocation-2/containers/task-assignment/task-assignment-container.component.ts

```diff
@@ -63,38 +67,40 @@ export class TaskAssignmentContainerComponent implements OnInit {
     defaultSortFieldName: 'dueDate',
     fields: this.fields,
   };
-  public ngOnInit(): void {
-    // Set up the default sorting.
-    this.sortedBy = {
-      fieldName: this.taskServiceConfig.defaultSortFieldName,
-      order: this.taskServiceConfig.defaultSortDirection
-    };
 
+  public ngOnInit(): void {
+    this.initForm();
     // Get the task from the route, which will have been put there by the resolver.
     const task = this.route.snapshot.data.taskAndCaseworkers.task.task;
     const caseworkers = this.route.snapshot.data.taskAndCaseworkers.caseworkers;
     task.assigneeName = getAssigneeName(caseworkers, task.assignee);
-    this.tasks = [ task ];
+    this.tasks = [task];
     this.verb = this.route.snapshot.data.verb as TaskActionType;
     this.successMessage = this.route.snapshot.data.successMessage as InfoMessage;
     if (task.assignee) {
       const names: string[] = task.assignee.split(' ');
       const firstName = names.shift();
       const lastName = names.join(' ');
-      this.excludedCaseworkers = [ { firstName, lastName } as Caseworker ];
+      this.excludedCaseworkers = [{ firstName, lastName } as Caseworker];
     }
     if (task.location) {
       this.location = { locationName: task.location } as Location;
     }
   }
 
+  public initForm(): void {
+    this.formGroup = new FormGroup({
+      caseWorkerName: new FormControl()
+    });
+  }
+
   public assign(): void {
     if (!this.caseworker) {
       this.error = NAME_ERROR;
       return;
     }
     this.error = null;
-    this.taskService.assignTask(this.tasks[0].id, {userId: this.caseworker.idamId}).subscribe(() => {
+    this.taskService.assignTask(this.tasks[0].id, { userId: this.caseworker.idamId }).subscribe(() => {
       this.reportSuccessAndReturn();
```

Comment:
this will probably cause issues as the subscription block is triggered by a button click event, i.e. if you click the assign button for many times and it may create many subscriptions.

Reply:
I have not added this logic. This is just a change to the template for future reassign work

Reply:
I see it only changed the format, I think we need to refactor the component's subscription/unsubscription logic.

Reply:
changed